rectools/model_selection/cross_validate.py

```diff
@@ -129,6 +136,11 @@ def cross_validate(  # pylint: disable=too-many-locals
         interactions_df_test = interactions.df.iloc[test_ids]  # 1x internal
         test_users = interactions_df_test[Columns.User].unique()  # 1x internal
         catalog = interactions_df_train[Columns.Item].unique()  # 1x internal
+        if ref_models is not None and validate_ref_models:
```

Comment:
There is some misunderstanding of logic I think.

If ref_models is not None -> `ref_reco` dict needs to be collected from all ref models recommendations. Other models will receive intersection metric for their recommendations with these ref models recommendation.
If ref_models is None -> `ref_reco = None`

If `validate_ref_models` is set to False then ref models recommendations are only used for other models metrics. We do not compute metrics for these models themselves.
If `validate_ref_models` is set to True then ref models also get all metrics calculated apart from providing recommendations for intersections calculation.

I'm sorry if I didn't write it explicitly in the issue description. If you have any questions please feel free to ask them here or in telegram. I'll be happy to help.